src/Disks/IO/ReadBufferFromRemoteFSGather.cpp

```diff
@@ -155,15 +156,26 @@ bool ReadBufferFromRemoteFSGather::readImpl()
      * we save how many bytes need to be ignored (new_offset - position() bytes).
      */
     if (bytes_to_ignore)
+    {
         current_buf->ignore(bytes_to_ignore);
+        bytes_to_ignore = 0;
+    }
 
-    auto result = current_buf->next();
+    bool result = current_buf->hasPendingData();
+    if (result)
+    {
+        /// bytes_to_ignore alreade added.
+        absolute_position += current_buf->available();
```

Comment:
How `bytes_to_ignore` affect absolute position?

Reply:
`bytes_to_ignore` is included in `file_offset_of_buffer_end` when define value of `bytes_to_ignore`. Then we pass `file_offset_of_buffer_end` to `ReadBufferFromRemoteFSGather` and assign to `absolute_position` https://github.com/ClickHouse/ClickHouse/blob/2ae5eb4cfad581011a6d86fcc3a83a21fa8f3466/src/Disks/IO/ReadBufferFromRemoteFSGather.cpp#L76
so that is why `bytes_to_ignore` is already included in `absolute_position`, after that we do `ignore` and need to add only `available` to `absolute_position`. `absolute_position` is in total how far the file was read.